### ISSUE TITLE
Product by Attributes notices--schema update needed?

### ISSUE THREAD

**Issue by peterfabian** [woocommerce/woocommerce-gutenberg-products-block#2717]
**Describe the bug**
```
PHP Notice: Undefined index: items in /srv/www/wordpress-one/public_html/wp-includes/rest-api.php on line 1245
PHP Notice: Trying to access array offset on value of type null in /srv/www/wordpress-one/public_html/wp-includes/rest-api.php on line 1220
PHP Notice: Trying to access array offset on value of type null in /srv/www/wordpress-one/public_html/wp-includes/rest-api.php on line 1234
PHP Notice: Trying to access array offset on value of type null in /srv/www/wordpress-one/public_html/wp-includes/rest-api.php on line 1252
PHP Notice: Trying to access array offset on value of type null in /srv/www/wordpress-one/public_html/wp-includes/rest-api.php on line 1292
PHP Notice: Trying to access array offset on value of type null in /srv/www/wordpress-one/public_html/wp-includes/rest-api.php on line 1308
PHP Notice: Trying to access array offset on value of type null in /srv/www/wordpress-one/public_html/wp-includes/rest-api.php on line 1313
PHP Notice: Trying to access array offset on value of type null in /srv/www/wordpress-one/public_html/wp-includes/rest-api.php on line 1318
PHP Notice: Trying to access array offset on value of type null in /srv/www/wordpress-one/public_html/wp-includes/rest-api.php on line 1323
PHP Notice: Trying to access array offset on value of type null in /srv/www/wordpress-one/public_html/wp-includes/rest-api.php on line 1350
PHP Notice: Undefined variable: taxonomy in /srv/www/wordpress-one/public_html/wp-content/plugins/woocommerce/packages/woocommerce-blocks/src/StoreApi/Routes/ProductCollectionData.php on line 96

```

**To Reproduce**

1. Created a page with All products, Filter by price and Filter by attributes blocks.
2. Tried to filter by attribute Size_swatch (e.g. S)
3. Noticed a couple of Notices in the logs.

**Expected behavior**

**Screenshots**
Attribute settings:
<img width="1747" alt="Screenshot 2020-06-15 at 15 49 45" src="https://user-images.githubusercontent.com/2207451/84665280-f35c9780-af1f-11ea-9906-309b1a15be2d.png">
Debug view of first error:
![Screenshot 2020-06-15 at 15 40 44](https://user-images.githubusercontent.com/2207451/84665303-f788b500-af1f-11ea-90de-ad550fb47425.png)

**WordPress (please complete the following information):**

* Core version: 5.4.1
* WooCommerce version: master
* WooCommerce Blocks version: 2.7.0
* Site language: en_US
* Any other plugins installed: same as previous notice issues

Related to #2715, just some more notices from other part of code, so logged separately, please close the other one as duplicate if you prefer.

**Comment by Aljullu** [comment]
I can reproduce this one too, thanks for filling @peterfabian!

It doesn't seem to be a regression since I could reproduce in 2.5.x, but marking it as a bug.

**Comment by nerrad** [comment]
This would be another fix good to get into 2.7.1 so assigning to that milestone for now.
